torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -353,7 +353,9 @@ void Reducer::check_grad_layout(
   }
 }
 
-void Reducer::copy_grad_to_bucket(at::Tensor& grad, at::Tensor& bucket_view) {
+void Reducer::copy_grad_to_bucket(
+    const at::Tensor& grad,
```

Comment:
Adding `const` to a Tensor doesn't make much difference, you can still modify the Tensor despite it being const.

Reply:
Yes, I understand that the underlying data can still change despite of const.

However, at least "const" can guarantee the metadata fields of tensor untouched, which is still nice. If you check the function body here, both `at::native::mul_out `and `bucket_view.copy_` pass the input tensor by const reference, so why don't we make it consistent here?

I think "const" here can improve the readability, by indicating that this method don't intend to modify the input grad tensor, although the underlying data can be modified.